godtools/App/Features/AppLanguage/Data-DomainInterface/GetDownloadableLanguagesListRepository.swift

```diff
@@ -0,0 +1,123 @@
+                let languageNameInOwnLanguage = self.getTranslatedLanguageName.getLanguageName(
+                    language: language,
+                    translatedInLanguage: language.code
+                )
+                let languageNameInAppLanguage = self.getTranslatedLanguageName.getLanguageName(
+                    language: language,
+                    translatedInLanguage: currentAppLanguage
+                )
+                
+                let toolsAvailableText = self.getToolsAvailableText(for: language.languageCode, translatedIn: currentAppLanguage)
+                
+                let downloadStatus = self.getDownloadStatus(for: language.id)
+                
+                return DownloadableLanguageListItemDomainModel(
+                    languageId: language.id,
+                    languageNameInOwnLanguage: languageNameInOwnLanguage,
+                    languageNameInAppLanguage: languageNameInAppLanguage,
+                    toolsAvailableText: toolsAvailableText,
+                    downloadStatus: downloadStatus
+                )
+            }
+            .sorted { language1, language2 in
+                
+                return self.getSortOrder(language1: language1, language2: language2)
+            }
+        }
+        .eraseToAnyPublisher()
+    }
+}
+
+// MARK: - Private
+
+extension GetDownloadableLanguagesListRepository {
+    
+    private func getToolsAvailableText(for languageCode: String, translatedIn translationLanguage: AppLanguageDomainModel) -> String {
+        
+        let filter = ResourcesFilter(
+            category: nil,
+            languageCode: languageCode,
+            resourceTypes: ResourceType.toolTypes
+        )
+        
+        let numberOfTools = resourcesRepository.getCachedResourcesByFilter(filter: filter).count
+        let localeId = translationLanguage
+        
+        let formatString = localizationServices.stringForLocaleElseSystemElseEnglish(
+            localeIdentifier: localeId,
+            key: ToolStringKeys.ToolFilter.toolsAvailableText.rawValue,
+            fileType: .stringsdict
+        )
+        
+        return String.localizedStringWithFormat(formatString, numberOfTools)
+    }
+    
+    private func getDownloadStatus(for languageId: String) -> LanguageDownloadStatusDomainModel {
+        
+        guard let downloadedLanguage = downloadedLanguagesRepository.getDownloadedLanguage(languageId: languageId) else {
+            
+            return .notDownloaded
+        }
+        
+        if downloadedLanguage.isDownloaded {
+            
+            return .downloaded
+            
+        } else {
+            
+            return .downloading(progress: downloadedLanguage.downloadProgress)
+        }
+    }
+    
+    private func getSortOrder(language1: DownloadableLanguageListItemDomainModel, language2: DownloadableLanguageListItemDomainModel) -> Bool {
+        
+        if language1.isDownloaded && !language2.isDownloaded {
```

Comment:
Hey @rachaelblue I'm wondering if here we should be sorting those by when the language was downloaded.  We do track that ```createdAt: Date``` in DownloadedLanguageDataModel to know when a language was downloaded. That way the order would remain consistent.
I didn't see in Figma if it should be ascending or descending.  If they want most latest downloaded at the top or the first downloaded at the top.

Reply:
Hey @rachaelblue let's leave this sorting as is, all it says in the Figma is that downloaded languages should be at the top.  Going to confirm with Jill if there should be any extra sorting on those and that can be done in a separate ticket if needed.  The only concern I have is that the order of those downloaded languages at the top could appear random since it would depend on how languages are fetched from realm.

Reply:
Ok yeah, the downloaded languages at the top will still sort alphabetically relative to each other, the way I have it right now.  But Figma doesn't show them alphabetically, so maybe it is by timestamp?  Would be good to check with Jill.

Reply:
Okay nice I didn't realize those were sorting alphabetically.  Yeah double check with Jill.  Just as long as they are sorting in some manner would be preferable that way the list stays consistent each time a user taps into it.

Reply:
Jill said sort alphabetically, so we're good 👍

Reply:
Awesome thanks @rachaelblue